### PR TITLE
[CI][SYCL-CTS] Ensure link/test runs are parallel

### PR DESCRIPTION
**Pull request by aelovikov-intel**
https://github.com/intel/llvm/pull/13421 reduced parellization when trying to avoid early build failouts. Use `ninja`'s `-k0` to do the same without undesired side effects. Also run individual tests in parallel as well.

### CODE REVIEW

**Review by KornevNikita — APPROVED** [review]
Definitely much better, but it would be helpful to have a list of failed to build categories in the summary. Anyway, it may be tricky, so we can left it for the future.

**Comment by KornevNikita** [comment]
BTW - there is a "test_conformance" target to build all except `test_all`: https://github.com/KhronosGroup/SYCL-CTS/commit/1aba52f831c79ac7fa958cb5006ee70d4a4bf73c

**Comment by aelovikov-intel** [comment]
Manual testing runs:
https://github.com/intel/llvm/actions/runs/8820525882/job/24214350497 (just two tests)
https://github.com/intel/llvm/actions/runs/8820553122 (default full run)

File: .github/workflows/sycl-linux-run-tests.yml [hunk 1]
```diff
@@ -118,7 +118,7 @@ on:
           Suggested variables: for E2E tests - LIT_FILTER, LIT_FILTER_OUT.
           LIT_OPTS won't work as we redefine it as part of this workflow.
 
-          For SYCL CTS - CTS_TEST_ARGS that will be passed as an argument to the
+          For SYCL CTS - CTS_TESTS_TO_BUILD that will be passed as an argument to the
           test_all executable.
```
Comment (KornevNikita) [inline .github/workflows/sycl-linux-run-tests.yml:122]: "...as an argument to 'ninja'" or something like that.

**Review by KornevNikita — APPROVED** [review]

File: .github/workflows/sycl-linux-run-tests.yml (hunk 1 above)
Comment (aelovikov-intel) [reply .github/workflows/sycl-linux-run-tests.yml:122]: I thought I've fixed this...

**Comment by aelovikov-intel** [comment]
@intel/dpcpp-devops-reviewers this is ready to your review.

**Comment by aelovikov-intel** [comment]
> @intel/dpcpp-devops-reviewers this is ready for your review.

@steffenlarsen @cperkinsintel @uditagarwal97 @intel/dpcpp-devops-reviewers , ping x2.

**Comment by uditagarwal97** [comment]
https://github.com/intel/llvm/actions/runs/8821781116/job/24220570301?pr=13533
I see that SYCL CTS Build and test jobs are skipped. Is that intentional?

**Comment by aelovikov-intel** [comment]
> https://github.com/intel/llvm/actions/runs/8821781116/job/24220570301?pr=13533 I see that SYCL CTS Build and test jobs are skipped. Is that intentional?

Yes, we only run it in Nightly.

**Review by uditagarwal97 — APPROVED** [review]
LGTM!

**Comment by aelovikov-intel** [comment]
Failures are unrelated - they are on e2e and not on cts.

**Pull request merged**